facebook/docusaurus
# fix(cssnano-preset): disable z-index minification

## Pre-flight checklist

- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #7548) and the maintainers have approved on my working plan.



## Motivation

`z-index` minification can cause issues when working with the styles of third party libraries. 

## Test Plan

On the [dogfooding page](https://deploy-preview-7593--docusaurus-2.netlify.app/tests/pages/z-index-tests/), the text would normally have its ` z-index` minified to 3, but after these changes, it remains 100.

For comparison, one place on the Docusaurus website that already uses `z-index` is on the [showcase page](https://docusaurus.io/showcase)
1. Go to https://docusaurus.io/showcase and open Chrome DevTools with responsive dimensions and device set to Desktop
2. Hover over a tag (e.g. "Favorite" or "Open-Source") and then right click on it
3. If you navigate to the element in DevTools (it should have the role "tooltip")
4. On the [official site](https://docusaurus.io/showcase) the z-index will be 7, but on the [site deploy of this pull request](https://deploy-preview-7593--docusaurus-2.netlify.app/showcase), the z-index is 500, which is what it is defined as in the CSS file.

### Test links



Dogfooding: https://deploy-preview-7593--docusaurus-2.netlify.app/tests/pages/z-index-tests/
Deploy preview: https://deploy-preview-7593--docusaurus-2.netlify.app/

## Related issues/PRs



#7548
